src/main/java/com/es/phoneshop/web/ProductDemodataServletContextListener.java

```diff
@@ -0,0 +1,49 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.PriceHistory;
+import com.es.phoneshop.model.product.Product;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+
+public class ProductDemodataServletContextListener implements ServletContextListener {
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        String contextParameter = servletContextEvent.getServletContext().getInitParameter("getProducts");
+        if (Boolean.valueOf(contextParameter)) {
+            getSampleProducts();
+        }
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
+
+    }
+
+    private void getSampleProducts() {
+        ArrayList<PriceHistory> priceHistories = new ArrayList<>();
+        priceHistories.add(new PriceHistory("1 Set 2018", new BigDecimal(100)));
+        priceHistories.add(new PriceHistory("10 Oct 2018", new BigDecimal(110)));
+        priceHistories.add(new PriceHistory("10 Jan 2019", new BigDecimal(150)));
+        ArrayListProductDao result = ArrayListProductDao.getInstance();
```

Comment:
https://medium.com/@sanchezdale/oop-good-practices-coding-to-the-interface-baea84fd60d3